test/cmd/images_tests.sh

```diff
@@ -51,6 +51,34 @@ os::cmd::try_until_success 'oc get imagestreamtags wildfly:latest'
 os::cmd::expect_success_and_text "oc get imageStreams wildfly --template='{{ index .metadata.annotations \"openshift.io/image.dockerRepositoryCheck\"}}'" '[0-9]{4}\-[0-9]{2}\-[0-9]{2}' # expect a date like YYYY-MM-DD
 os::cmd::expect_success_and_text 'oc get istag' 'wildfly'
 
+# https://github.com/openshift/origin/issues/10402
+# Make sure that switching from a reference tag to a non-reference tag properly
+# updates the generation.
+# We will move the 'latest' tag to point at the same reference as '8.1'.
+orig_spec_tag_generation="$(oc get imagestreams/wildfly --template='{{range .spec.tags}}{{ if eq .name "latest"}}{{ .generation }}{{end}}{{end}}')"
+orig_status_tag_generation="$(oc get imagestreams/wildfly --template='{{range .status.tags}}{{ if eq .tag "latest"}}{{ $item:=index .items 0}}{{$item.generation}}{{end}}{{end}}')"
+reference_81="$(oc get imagestreams/wildfly --template='{{range .status.tags}}{{ if eq .tag "8.1"}}{{ $item:=index .items 0}}{{$item.dockerImageReference}}{{end}}{{end}}')"
+
+# https://github.com/openshift/origin/issues/10402
+# Make latest a --reference to 8.1
+os::cmd::expect_success 'oc tag --source=docker $reference_81 --reference wildfly:latest'
+sleep 5
```

Comment:
See the comment above; I am trying to find a way to use the `os::cmd::try_until_{text,success,failure}` routines, but I was having trouble and hence asking for advice in this ticket.

I think I had another idea this morning, so I'm going to try that and see if it works.